src/cli/index.js

```diff
@@ -583,7 +584,11 @@ export async function main({
         reporter.info(command.getDocsInfo);
       }
 
-      return exit(1);
+      if (err instanceof ProcessTermError) {
+        return exit(err.EXIT_CODE || 1);
+      }
+
+      return exit((err.code && Number(err.code)) || 1);
```

Comment:
This is not pretty, but when we convert the `ProcessTermError` down to `MessageError`, we lost the `EXIT_CODE` as well.

https://github.com/yarnpkg/yarn/pull/6926/files#diff-939cc0d319704d4f0d98baf9167ba850